test/built-ins/decodeURI/throws-URIError.js

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2025 ayuan.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+info: |
+    Verify decodeURIComponent throws URIError for various invalid UTF-8 sequences
+    Reference: https://stackoverflow.com/a/1319229/172999
+esid: sec-decodeuricomponent-encodeduricomponent
+description: |
+    Invalid sequences include:
+    - Surrogate pair encoding
+    - Overlong encoding
+    - Invalid continuation bytes
+    - Incomplete sequences
+    - Out-of-range code points
+features: [uricode]
+---*/
+
+// CHECK#1: Reserved surrogate pair (U+D800-DFFF)
+(function CHECK1() {
+    var result = true;
+    try {
+        decodeURIComponent('%ED%BF%BF');
+        result = false;
+    } catch (e) {
+        if (!(e instanceof URIError)) result = false;
+    }
+    if (!result) {
+        throw new Test262Error('#1: %ED%BF%BF (surrogate pair) should throw URIError');
+    }
+})();
```

Comment:
It'd be better to use the built-in `assert.throws` for these checks, just for consistent error reporting. For example:
```js
assert.throws(URIError, function CHECK1() {
  decodeURIComponent('%ED%BF%BF');
}, '#1: %ED%BF%BF (surrogate pair) should throw URIError');
```